src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -26,5 +28,44 @@ private static bool GetGlobalizationInvariantMode()
             }
             return invariantEnabled;
         }
+
+        private static bool LoadIcu()
+        {
+            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            {
+                return Interop.Globalization.LoadICU() != 0;
+            }
+
+#if TARGET_OSX
+            string extension = ".dylib";
+            bool versionAtEnd = false;
+#else
+            string extension = "so.";
```

Comment:
>so [](start = 32, length = 2)

what happen when we don't have version? this extension will be wrong, right?

Reply:
Yeah, I didn't add support for no-version. I wonder what the format of the switch should be for non-versioned app-local ICU?

Reply:
":"

Reply:
Ok, sounds good.